modules/veteran/app/controllers/veteran/v0/other_accredited_representatives_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Veteran
+  module V0
+    class OtherAccreditedRepresentativesController < BaseAccreditedRepresentativesController
+      before_action :verify_type
+
+      PERMITTED_TYPE = %w[attorney].freeze
+
+      private
+
+      def serializer_class
+        'Veteran::Accreditation::BaseRepresentativeSerializer'.constantize
+      end
+
+      def representative_query
+        query = base_query.select("veteran_representatives.*, #{distance_query_string}").where(
+          '? = ANY(user_types) ', search_params[:type]
+        )
+
+        search_params[:name] ? find_with_name_similar_to(query) : query
+      end
+
+      def find_with_name_similar_to(query)
```

Comment:
I don't know how difficult it would be, but I suggest looking into moving the two name search methods to the representative model or to a service. It will make testing and implementing changes to the search more straightforward and rigorous.

Reply:
Agreed we'll address this when we move the code into our own module